src/virtualkeyboardlayout.h

```diff
@@ -35,6 +35,17 @@ class KeyboardLayout {
         }
     }
 
+    ~KeyboardLayout() {
+        FCITX_KEYBOARD_LAYOUT() << "~KeyboardLayout()";
+        keys_.clear();
+        stateLabels_.clear();
+        modeLabels_.clear();
+        modeActions_.clear();
+        modeOffsets_.clear();
+        // decrement the reference count
+        json_object_put(json_);
+        free(json_);
```

Comment:
Is it O.K. to pass it to `free()`?
[According to the document](https://json-c.github.io/json-c/json-c-0.10/doc/html/json__object_8h.html#accc34b38e15e46357d64a0beed3d0150), `json_object_put()` does it when the reference count is decremented to `0`.

Reply:
It seems  that no need to free here. Thanks.